tests/openssh.rs

```diff
@@ -282,324 +350,345 @@ macro_rules! assert_kind {
+            .raw_args(std::iter::once("1 2"))
+            .output()
+            .await
+            .unwrap())
+        .status;
+        assert!(status.success());
+
+        let status = dbg!(session
+            .raw_command("printf '%d %d'")
+            .arg("1")
+            .arg("2")
+            .output()
+            .await
+            .unwrap())
+        .status;
+        assert!(status.success());
 
-    session.close().await.unwrap();
+        session.close().await.unwrap();
+    }
 }
 
 #[tokio::test]
 #[cfg_attr(not(ci), ignore)]
 async fn process_exit_on_signal() {
-    let session = Session::connect(&addr(), KnownHosts::Accept).await.unwrap();
-
-    let mut sleeping = session.command("sleep").arg("5566").spawn().unwrap();
-
-    // give it some time to make sure it starts
-    tokio::time::sleep(std::time::Duration::from_secs(1)).await;
-
-    // Now stop that process.
-    //
-    // We use `pkill -f` to match on the number rather than the `sleep` command, since other tests
-    // may use `sleep`. We use `-o` to ensure that we don't accidentally kill the ssh connection
-    // itself, but instead match the _oldest_ matching command.
-    let killed = session
-        .command("pkill")
-        .arg("-f")
-        .arg("-o")
-        .arg("5566")
-        .output()
-        .await
-        .unwrap();
-    eprintln!("{:?}", killed);
-    assert!(killed.status.success());
-
-    // await that process — this will yield "Disconnected", since the remote process disappeared
-    let failed = sleeping.wait().await.unwrap_err();
-    eprintln!("{:?}", failed);
-    assert!(matches!(failed, Error::Disconnected));
-
-    // the connection should still work though
-    let _ = session.check().await.unwrap();
+    for session in connects().await {
+        let mut sleeping = session.command("sleep").arg("5566").spawn().await.unwrap();
+
+        // give it some time to make sure it starts
+        tokio::time::sleep(std::time::Duration::from_secs(1)).await;
+
+        // Now stop that process.
+        //
+        // We use `pkill -f` to match on the number rather than the `sleep` command, since other tests
+        // may use `sleep`. We use `-o` to ensure that we don't accidentally kill the ssh connection
+        // itself, but instead match the _oldest_ matching command.
+        let killed = session
+            .command("pkill")
+            .arg("-f")
+            .arg("-o")
+            .arg("5566")
+            .output()
+            .await
+            .unwrap();
+        eprintln!("process_exit_on_signal: {:?}", killed);
+        assert!(killed.status.success());
+
+        // await that process — this will yield "Disconnected", since the remote process disappeared
+        eprintln!("process_exit_on_signal: Waiting for sleeping to exit");
+        let failed = sleeping.wait().await.unwrap_err();
+        eprintln!("process_exit_on_signal: {:?}", failed);
+        assert!(matches!(failed, Error::RemoteProcessTerminated));
```

Comment:
Why did this change?

Reply:
Revert unnecessary changes to this code.

However, `Error::RemoteProcessTerminated` is still there for I change the `process_impl::RemoteChild` to return `RemoteProcessTerminated` on exit code 255.

I also did the same thing in `mux_client_impl::RemoteChild`, return `RemoteProcessTerminated` if the connection is severed without any exit code.

Reply:
Hmm, that's odd to me — I thought exit 255 didn't _necessarily_ mean that the remote process terminated?

Reply:
I am not sure about that.
I modified that to fix the integration test.

Maybe it should call session.check() here?

Reply:
I don't _think_ that should be necessary. After all, it's the remote process that exited here — the connection is still fine. I suppose `RemoteProcessTerminated` is the correct error code here, it just makes me wonder if we then also start returning `RemoteProcessTerminated` if the remote process _didn't_ fail and instead it was just the connection that failed?

Reply:
I'm not sure about that, these kind of scenario is not tested and is probably hard to setup, which is why I recommended calling `session.disconnect()` to check for connection problem.

It could even be that the ssh multiplex master is killed by something else.

Reply:
However, there is also a race condition if we call `session.disconnect()` after the `wait`...

Reply:
TBH, I cannot think of a reliable way of telling whether exit code 255 is `RemoteProcessTerminated` or `Disconnected`.

The only thing I noticed that can be used to identify `Disconnected` is that ssh multiplex master would print `Shared connection to localhost closed.` if it is disconnected.

However, due to the fact that the `stdout` and `stderr` can be captured by the users of this crate, I think the only way is to document this behavior.

Reply:
Doc for `RemoteChild::{wait, try_wait}` updated.